boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,212 @@
+        'price_details': jsonify_price(Price.query.filter(Price.discount_policy == policy).first()) if policy.is_price_based else '',
+        'currency': CURRENCY_SYMBOL['INR'],
+        'dp_items': [{'id': str(item.id), 'title': item.title} for item in policy.items]
+    }
+
+
+def jsonify_discount_policies(data_dict):
+    discount_policies_list = []
+    for policy in data_dict['discount_policies']:
+        discount_policies_list.append(jsonify_discount_policy(policy))
+    return jsonify(
+        org_name=data_dict['org'].name, title=data_dict['org'].title,
+        discount_policies=discount_policies_list,
+        total_pages=data_dict['total_pages'],
+        paginated=data_dict['total_pages'] > 1,
+        current_page=data_dict['current_page']
+    )
+
+
+@app.route('/api/1/admin/<org>/discount_policy')
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    permission='org_admin'
+    )
+@render_with({'text/html': 'index.html', 'application/json': jsonify_discount_policies}, json=True)
+@requestargs('search', ('page', int))
+def admin_discount_policies(organization, search=None, page=1):
+    results_per_page = 20
+
+    if request.is_xhr:
+        discount_policies = organization.discount_policies
+
+        if search:
+            discount_policies = discount_policies.filter(
+                DiscountPolicy.title.ilike('%{query}%'.format(query=search)))
+
+        total_policies = discount_policies.count()
+        total_pages = int(math.ceil(total_policies / results_per_page))
+        offset = (page - 1) * results_per_page
+
+        discount_policies = discount_policies.limit(results_per_page).offset(offset).all()
+
+        return dict(
+            org=organization, title=organization.title,
+            discount_policies=discount_policies,
+            total_pages=total_pages,
+            paginated=(total_policies > results_per_page),
+            current_page=page)
+    else:
+        return dict(org=organization, title=organization.title)
+
+
+@app.route('/api/1/admin/<org>/discount_policy/new', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_add_discount_policy(organization):
+    discount_policy_form = DiscountPolicyForm()
+    discount_policy_form.items.query = Item.query.join(ItemCollection).filter(ItemCollection.organization == organization).options(db.load_only('id')).all()
+    if not discount_policy_form.validate_on_submit():
+        return api_error(message=discount_policy_form.errors, status_code=400)
+    discount_policy = DiscountPolicy(title=discount_policy_form.title.data, organization=organization)
+    discount_policy_form.populate_obj(discount_policy)
+    items = request.form.getlist('items')
+    for item_id in items:
+        item = Item.query.get(item_id)
```

Comment:
I suggest renaming `items` to `item_ids` since it's a list of item ids. This can also be done a little more efficiently. You can fetch all the items at once: `items = Item.query.filter(Item.id.in_(item_ids)).`